jetty-client/src/main/java/org/eclipse/jetty/client/HttpDestination.java

```diff
@@ -439,25 +459,27 @@ public void release(Connection connection)
 
     public boolean remove(Connection connection)
     {
-        return connectionPool.remove(connection);
-    }
-
-    public void close(Connection connection)
-    {
-        boolean removed = remove(connection);
+        boolean removed = connectionPool.remove(connection);
 
         if (getHttpExchanges().isEmpty())
         {
             tryRemoveIdleDestination();
         }
         else
         {
-            // We need to execute queued requests even if this connection failed.
-            // We may create a connection that is not needed, but it will eventually
-            // idle timeout, so no worries.
+            // We need to execute queued requests
+            // even if this connection was removed.
+            // We may create a connection that is not
+            // needed, but it will eventually idle timeout.
             if (removed)
-                process();
+                process(true);
         }
+        return connectionPool.remove(connection);
+    }
+
+    public void close(Connection connection)
```

Comment:
Since `close(Connection)` and `remove(Connection)` are now identical, shouldn't one of them be removed?

Reply:
I deprecated `close(Connection)` to be removed in 10.